wsmaster/che-core-api-workspace-activity/src/test/java/org/eclipse/che/api/workspace/activity/WorkspaceActivityManagerTest.java

```diff
@@ -106,4 +114,80 @@ public void shouldCeaseToTrackTheWorkspaceActivityAfterStopping() throws Excepti
 
     verify(workspaceActivityDao, times(1)).removeExpiration(eq(wsId));
   }
+
+  @Test
+  public void shouldRecordWorkspaceCreation() throws Exception {
+    String wsId = "1";
+
+    EventSubscriber<WorkspaceCreatedEvent> subscriber = subscribeAndGetCreatedSubscriber();
+
+    subscriber.onEvent(
+        new WorkspaceCreatedEvent(
+            DtoFactory.newDto(WorkspaceDto.class)
+                .withId(wsId)
+                .withAttributes(ImmutableMap.of(Constants.CREATED_ATTRIBUTE_NAME, "15"))));
+
+    verify(workspaceActivityDao, times(1)).setCreatedTime(eq(wsId), eq(15L));
+  }
+
+  @Test(dataProvider = "wsStatus")
+  public void shouldRecordWorkspaceStatusChange(WorkspaceStatus status) throws Exception {
+    String wsId = "1";
+
+    EventSubscriber<WorkspaceStatusEvent> subscriber = subscribeAndGetStatusEventSubscriber();
+
+    subscriber.onEvent(
+        DtoFactory.newDto(WorkspaceStatusEvent.class).withStatus(status).withWorkspaceId(wsId));
+
+    verify(workspaceActivityDao, times(1)).setStatusChangeTime(eq(wsId), eq(status), anyLong());
+  }
+
+  @Test
+  public void shouldRemoveActivityWhenWorkspaceRemoved() throws Exception {
+    String wsId = "1";
+
+    EventSubscriber<BeforeWorkspaceRemovedEvent> subscriber = subscribeAndGetRemoveSubscriber();
+
+    subscriber.onEvent(
+        new BeforeWorkspaceRemovedEvent(
+            new WorkspaceImpl(DtoFactory.newDto(WorkspaceDto.class).withId(wsId), null)));
+
+    verify(workspaceActivityDao, times(1)).removeActivity(eq(wsId));
+  }
+
+  @DataProvider(name = "wsStatus")
+  public Object[][] getWorkspaceStatus() {
+    return Stream.of(WorkspaceStatus.values())
+        .map(s -> new WorkspaceStatus[] {s})
+        .toArray(Object[][]::new);
+  }
+
+  private EventSubscriber<WorkspaceStatusEvent> subscribeAndGetStatusEventSubscriber() {
+    subscribeToEventService();
+    @SuppressWarnings("unchecked")
+    final EventSubscriber<WorkspaceStatusEvent> subscriber =
+        (EventSubscriber<WorkspaceStatusEvent>) createAndChangeEventCaptor.getAllValues().get(0);
+    return subscriber;
+  }
+
+  private EventSubscriber<WorkspaceCreatedEvent> subscribeAndGetCreatedSubscriber() {
+    subscribeToEventService();
+    @SuppressWarnings("unchecked")
```

Comment:
Would not allow having two separate events captors for Create and Update Workspace Events avoid using casting, getting by element index and SuppressWarnings?